### PR TITLE
[WIP] more generic concept of utility...

### PR DESCRIPTION
**Pull request by jsenellart**
…to execute other dockerized tasks (word alignment, similarity checking, etc)...

+ some  style tests (enforced by pycodestyle/tox.ini)

### CODE REVIEW

File: nmtwizard/framework.py [hunk 1]
```diff
@@ -15,24 +15,25 @@
...
 logger = get_logger(__name__)
 
 
 @six.add_metaclass(abc.ABCMeta)
-class Framework(object):
+class Framework(Utility):
     """Base class for frameworks."""
 
     def __init__(self, stateless=False, support_multi_training_files=False):
+        super(Framework, self).__init__()
```
Comment (guillaumekln) [inline nmtwizard/framework.py]: Move this after the comment section.

File: nmtwizard/framework.py [hunk 2]
```diff
@@ -261,41 +243,35 @@ def run(self, args=None):
...
+
         if args.model_storage_read is None:
             args.model_storage_read = args.model_storage
         if args.model_storage_write is None:
             args.model_storage_write = args.model_storage
-        if (not self._stateless
-            and (args.cmd != 'preprocess' or args.build_model)
-            and (args.model_storage_write is None or args.model_storage_write is None)):
+
+        self._model_storage = args.model_storage
```
Comment (guillaumekln) [inline nmtwizard/framework.py]: This attribute should not be needed.

File: nmtwizard/framework.py (hunk 1 above)
Comment (jsenellart) [reply nmtwizard/framework.py]: done

File: nmtwizard/framework.py (hunk 2 above)
Comment (jsenellart) [reply nmtwizard/framework.py]: indeed, thanks

**Comment by guillaumekln** [comment]
Looks good to me. Please remove the [WIP] tag when it's OK for you (not sure if you are planning other changes).

**Comment by jsenellart** [comment]
thanks - I am finalizing the 2 first utilities:

* [similarity utility](/OpenNMT/nmt-wizard-docker/tree/dev/similarity_utility)
* [wordalign utility](/OpenNMT/nmt-wizard-docker/tree/dev/wordalign_utility)

and we can merge when everything works together. Note that I have now moved *all* the framework generic options to utility. Wordalign can indeed apply on a preprocess model and will fetch the model if necessary.

**Pull request merged**